app/components/maps/SidebarCharts.tsx

```diff
@@ -0,0 +1,385 @@
+                                }
+                            }
+                        });
+                        if (matchingPts.length) {
+                            chart.tooltip.refresh(matchingPts);
+                        }
+                    }
+                });
+                if (!validCharts) {
+                    clearInterval(globalInterval);
+                }
+            }
+        }, 1);
+    } else {
+        clearInterval(globalInterval);
+    }
+
+    addChartFunc(highCharts);
+    return (
+        highCharts
+    );
+};
+
+// eslint-disable-next-line no-undef
+let timer: NodeJS.Timeout;
+
+const debounce = (callback: () => any, timeout: number) => () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+        callback();
+    }, timeout);
+};
+
+interface Props {
+    replaysData: ReplayData[];
+    timeLineGlobal: TimeLineInfos;
+}
+export const SidebarCharts = ({
+    replaysData,
+    timeLineGlobal,
+}: Props): JSX.Element => {
+    const [visible, setVisible] = useState<boolean>(false);
+    const [syncWithTimeLine, setSyncWithTimeLine] = useState<boolean>(true);
+    const [selectedCharts, setSelectedCharts] = useState<ChartType[]>([]);
+    const {
+        numColorChange,
+    } = useContext(SettingsContext);
+
+    // Get each different currentRaceTime for all replays to get matching chart tooltips
+    const allRaceTimes: number[] = [];
+
+    replaysData.forEach((replay: ReplayData) => {
+        replay.samples.forEach((sample: ReplayDataPoint) => {
+            if (!allRaceTimes.includes(sample.currentRaceTime)) {
+                allRaceTimes.push(sample.currentRaceTime);
+            }
+        });
+    });
+    allRaceTimes.sort((a, b) => a - b);
+
+    let childCharts: any[] = [];
+
+    const addChart = (chart: any) => {
+        childCharts.push(chart);
+    };
+
+    let el1: any;
+    let isResizing = false;
+
+    const onClose = () => {
+        setVisible(false);
+    };
+
+    const toggleSidebar = () => {
+        setVisible(!visible);
+    };
+
+    const onMouseDown = (e: any) => {
+        isResizing = true;
+    };
+
+    const onMouseUp = (e: any) => {
+        isResizing = false;
+    };
+
+    const onMouseMove = (e: any) => {
+        if (isResizing) {
+            const offsetBottom = document.body.offsetHeight - (e.clientY - document.body.offsetTop);
+            const minHeight = 50;
+            const maxHeight = 800;
+            if (offsetBottom > minHeight && offsetBottom < maxHeight && el1) {
+                el1.style.height = `${offsetBottom}px`;
+            }
+        }
+    };
+
+    useEffect(() => {
+        el1 = document.querySelector('.ChartDrawer');
+
+        document.addEventListener('mousemove', onMouseMove);
+        document.addEventListener('mouseup', onMouseUp);
+
+        return () => {
+            document.removeEventListener('mousemove', onMouseMove);
+            document.removeEventListener('mouseup', onMouseUp);
+        };
+    });
```

Comment:
```suggestion
    }, []);
```
In this case you should add a `, []` as a second parameter. Without this second parameter, this piece of code will be run on every render.